pkg/acsengine/defaults.go

```diff
@@ -646,14 +647,15 @@ func setAgentProfileDefaults(a *api.Properties, isUpgrade, isScale bool) {
 			profile.AcceleratedNetworkingEnabled = helpers.PointerToBool(!isUpgrade && !isScale && helpers.AcceleratedNetworkingSupported(profile.VMSize))
 		}
 
-		// don't default Distro for OpenShift
-		if !a.OrchestratorProfile.IsOpenShift() {
-			if profile.Distro == "" {
+		if profile.Distro == "" {
+			if a.OrchestratorProfile.IsKubernetes() {
 				if profile.OSDiskSizeGB != 0 && profile.OSDiskSizeGB < api.VHDDiskSizeAKS {
 					profile.Distro = api.Ubuntu
 				} else {
 					profile.Distro = api.AKS
 				}
+			} else if !a.OrchestratorProfile.IsOpenShift() {
```

Comment:
@CecileRobertMichon we're covering the OpenShift condition here

Reply:
I know, my comment was regarding Tariq's proposed implementation

Reply:
Cool then. Thanks for the clarifications